crates/driver/src/domain/eth/mod.rs

```diff
@@ -11,7 +11,7 @@ mod gas;
 pub use {
     allowance::Allowance,
     eip712::{DomainFields, DomainSeparator},
-    ethcontract::PrivateKey,
+    ethcontract::{transaction::kms::Account as KmsAccount, Account, PrivateKey},
```

Comment:
Exposing AWS KMS details from the `domain::eth` module is not the correct design decision.

The `domain::eth` module is supposed to only contain definitions related to the Ethereum blockchain which constitute domain logic for our application.

OTOH, AWS KMS is an external key management service. It has nothing to do with Ethereum per se, and it's also an infrastructural (external) component, not part of the domain logic.

The root of the issue is that `ethcontract` crate bakes in the KMS logic instead of providing a more generic way to sign things (e.g. via callbacks or traits).

Reply:
Rather than re-exporting, in this case it might be worth it to have a separate definition of `domain::eth::Account` which doesn't expose any KMS-specific details and then map this type into `ethcontract::Account` somewhere in the `infra` module.

Reply:
Alternatively, since the `Account` type doesn't seem to be used by the `domain` logic at all (singing transactions is defined to be an external concern), you can avoid touching the `domain` entirely and have `infra::Solver::account` directly return `ethcontract::Account` - this is completely fine for now.